Parser/Tree_Nodes/stmt.c

```diff
@@ -63,15 +64,22 @@ stmt * create_stmt(GArray * token_stream, unsigned long index, unsigned long * n
                     case jdouble:
                     case jstring:
                         // Expression
-                        new_statement->expression = create_expr(token_stream, index, next);
+                        curToken = &g_array_index(token_stream, Token, *next);
+                        Type t;
+                        if (findIDType(curToken->data, &t)) {
+                            new_statement->re_asmt = create_r_asmt(token_stream, index, next);
+                        } else {
+                            new_statement->expression = create_expr(token_stream, index, next);
+                        }
```

Comment:
Not sure what is going on here. If the first token is an ID then it MUST be a reassign. If the first token is a type ("Integer", "Double", "String") then it is an assignment. Do the checking for ID existaning in the reasssignment constructor.